pkg/pub_worker/lib/src/bin/dartdoc_wrapper.dart

```diff
@@ -85,14 +85,39 @@ Future<void> postProcessDartdoc({
     _log.info('Finished .tar.gz archive');
   });
 
-  await Future.wait([
-    processingFuture,
-    archiveFuture,
-  ]);
+  try {
+    await Future.wait([
+      processingFuture,
+      archiveFuture,
+    ]);
 
-  // Move from temporary output directory to final one, ensuring that
-  // documentation files won't be present unless all files have been processed.
-  // This helps if there is a timeout along the way.
-  await Directory(tmpOutDir).rename(p.join(outputFolder, 'doc'));
-  await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+    // Move from temporary output directory to final one, ensuring that
+    // documentation files won't be present unless all files have been processed.
+    // This helps if there is a timeout along the way.
+    await tmpOutDir.rename(p.join(outputFolder, 'doc'));
+    await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+  } catch (e, st) {
+    // We are likely out of free disk space.
+    //
+    // One of the isolate has already stopped, the other may be still working.
+    // By deleting the dartdoc-generated content, the post-processing of the other
+    // isolate will stop too.
+    await Directory(docDir).deleteIgnoringErrors(recursive: true);
```

Comment:
I don't understand this.

Who owns the `docDir`, I think we should say that it's owned by the caller to `postProcessDartdoc` as is currently the case.

And that `postProcessDartdoc` only read from `docDir`.

It feels messy to be sometimes deleting it here and sometimes not.
If we want to delete it here, we should always do so, and we should document it for the caller.

However, I see no reason to do that.

--------------------------

Here we're trying to stop the isolates from running by deleting their input files.
I think that's a bit wild. Perhaps it'd be more reasonable to:
 * (A) Not run the two thing concurrently, so we don't have to use isolates.
 * (B) Create a `runInIsolate<T>(FutureOr<T> Function() computation, {Completer<void> cancel})`.

Obviously, (B) would require that we create a tiny helper library for running stuff in an isolate with cancellation. And that we have tests :rofl: 

I think the correct approach for what you're trying to do is (B), but that the simply solution is (A). Afterall, it's fairly likely that both these operations are going to be I/O bound (maybe not post-processing).

Reply:
Fixed the `docDir` deletion. Agreed on (B) being the better one, but for a simple handling we may just do the cleanup.